### PR TITLE
Make REST methods support redirects

### LINKED ISSUE #145: REST methods should handle redirects
### Scenario/Intent
The REST methods should handle redirects. One issue I've already run into is that for Synergy Enclsoures, trying to get the script (`enc.script`) gives a 301 error message saying it has moved to a LogicalEnclosure uri.

I think we could implements something pretty similar to [this](http://stackoverflow.com/a/6934503/2456552), where we only follow redirects up to 3 or 5 times.

Because we need the data to re-submit the request, we'll need to handle this in the [rest_api](https://github.com/HewlettPackard/oneview-sdk-ruby/blob/master/lib/oneview-sdk/rest.rb#L31) method, not response_handler.

### Environment Details
 - **OneView SDK Version:** 3.1.0
 - **OneView Appliance Version:** All
 - **OneView Client API Version:** All

### Steps to Reproduce
With a client using api_version 300 as `@client`, talking to a v300 DCS...

```ruby
@client.api_version = 300
OneviewSDK.api_version = 300
OneviewSDK::API300.variant = 'Synergy'

e = @client.get_all(:Enclosures).first
e.script
```

### Expected Result
It to be able to get the script (following any redirects needed to get there)

### Actual Result
I get this error:

```ruby
OneviewSDK::RequestError: 301
from /usr/local/rvm/gems/ruby-2.2.4/gems/oneview-sdk-3.1.0/lib/oneview-sdk/rest.rb:162:in `response_handler'
```

### PR DESCRIPTION
**Pull request by jsmartt**
### Description
Allows REST methods to support redirects. There's a default limit of 3 redirects, but is configurable.

### Issues Resolved
Fixes #145

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: spec/unit/rest_spec.rb [hunk 1]
```diff
@@ -37,6 +37,32 @@
       expect_any_instance_of(Net::HTTP).to receive(:open_timeout=).with(5).and_call_original
       client.rest_api(:get, path)
     end
+
+    it 'supports following redirects' do
+      response1 = FakeResponse.new({ name: 'New' }, 301, location: path)
+      response2 = FakeResponse.new({ name: 'New' }, 200, location: path)
+      allow_any_instance_of(Net::HTTP).to receive(:request).and_return(response1)
+      allow_any_instance_of(Net::HTTP).to receive(:request).and_return(response2)
```
Comment (fgbulsoni) [inline spec/unit/rest_spec.rb]: This line is overriding line 44. This tests is only checking if it returns response2 at the moment.

Comment (jsmartt) [inline spec/unit/rest_spec.rb]: Ah! I tried a few things and wasn't able to get it to order the returns. Do you know of a way to do this @fgbulsoni ?

Comment (fgbulsoni) [inline spec/unit/rest_spec.rb]: Using `expect` instead of `allow` should do the trick

Comment (tmiotto) [inline spec/unit/rest_spec.rb]: Or maybe using the `.with` clause and using different locations for each response.

Comment (jsmartt) [inline spec/unit/rest_spec.rb]: OK, it's not the prettiest thing, but I got it to work and properly mock the 301 followed by a 200 response.

Comment (ricardogpsf) [inline spec/unit/rest_spec.rb]: I think that work good doing this: [](https://github.com/rspec/rspec-mocks#consecutive-return-values).

**Pull request merged**